internal/handlers/common/filter.go

```diff
@@ -50,6 +50,21 @@ func FilterDocument(doc, filter *types.Document) (bool, error) {
 
 // filterDocumentPair handles a single filter element key/value pair {filterKey: filterValue}.
 func filterDocumentPair(doc *types.Document, filterKey string, filterValue any) (bool, error) {
```

Comment:
If I understand correctly, dot notation can be used at the deeper level too. For example, `{"$or": [{"value.foo": 42}, {"value.42": "foo"}]}` should be a valid query. But I did not check it :) If it works, then checking for dot should not be done there

Reply:
You are right, the example you provided is a valid query and it works for both Mongo and the proposed implementation -- I've just checked both

Reply:
I spent a significant amount of time trying to come up with an example that does not work – and could not. :) Great job!